proposals/0052-svg-components.md

```diff
@@ -0,0 +1,293 @@
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<Logo class="text-slate-500" />
+```
+
+This would generate the following output:
+
+```astro
+<svg width="24" height="24" role="img" class="text-slate-500" >
+    <!-- SVG Content -->
+</svg> 
+```
+
+### Sprite
+
+An SVG Sprite is when an `<svg>` element defines multiple SVGs (or symbols) using the `<symbol>` element and attaching a unique identifier to each SVG as an `id`. These are then able to be referenced using the `<use>` element with an `href` attribute that points to the `id` defined on the `<symbol>` element. 
+
+When a `.svg` file is first imported and rendered in Astro, the system will convert the file into a `<symbol>` element. This will be inserted into the `<svg>` element. This approach ensures that all subsequent renders of that `.svg` can use a `<use>` element, referencing the ID of the initial `<symbol>`.
+
+This method was chosen by `astro-icon` to optimize SVG usage by default. It is riskier because it does more manipulation to the `.svg` file and relies on the `id` reference being available. The benefit to this method is that we now only have a single definition of repeated usages of the same `.svg` file and only minimal code to reference the definition in other locations.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg';
+---
+
+<!-- First Usage -->
+<Logo class="text-gray-700" />
+
+<!-- Second Usage -->
+<Logo class="text-yellow-50" />
+
+```
+
+This would generate the following output:
+
+```astro
+<svg width="24" height="24" role="img" class="text-gray-700">
+    <symbol id="a:0">
+        <!-- SVG Content -->
+    </symbol>
+    <use href="#a:0" />
+</svg> 
+<svg width="24" height="24" role="img"  class="text-yellow-50">
+    <use href="#a:0" />
+</svg> 
+```
+
+## Prop Inheritance and Overriding
+
+When a `.svg` is treated as a component, users can pass attributes such as `width`, `height`, `fill`, `stroke`, and other common SVG attributes. These attributes will automatically be applied to the root `<svg>` element unless they conflict with existing properties in the original `.svg` file, in which case the passed props will override the existing ones.
+
+**Example:**
+
+```astro
+---
+import Logo from '../assets/logo.svg'
+---
+
+<Logo />
+
+<!-- Pass width and height to override default size -->
+<Logo width={100} height={100} fill="blue" />
+```
+
+This would generate the following output:
+
+```astro
+<svg width="24" height="24" role="img">
+    <!-- SVG Content -->
+</svg> 
+<svg width="100" height="100" role="img" fill="blue">
+    <!-- SVG Content -->
+</svg> 
+```
+
+### Sizing
+
+To simplify the process of setting the dimensions of SVG components, this proposal introduces a `size` prop that developers can use to uniformly scale the width and height of an SVG. This prop provides a convenient shorthand for situations where both width and height need to be set to the same value, offering a more concise API.
```

Comment:
I'm not sure if there's enough benefit to having this `size` prop. It adds some complexity, while saving users only a few characters of typing at most.

Three downsides I can think of:
- The types now diverge from `SVGAttributes`.
- Unexpected results when the SVG isn't intended to have a 1:1 aspect-ratio.
- Additional API for users to discover and learn.

I think a framework wrapper over SVG should stay as close to `<svg>` as possible.

Reply:
Appreciate the feedback! This was a carryover from astro-icon. We primarily had the use-case of square icons so this was an extremely useful. I'm open to removing it if we feel that it isn't super useful.

Reply:
This ties back to an unresolved question as well. 
https://github.com/withastro/roadmap/pull/1035/files#diff-ab3122e95078e177eab4f364380c44e582e73c496b98809d7d5e966f9d09e1f1R328